### PR TITLE
Quantizer returns ids as int32, not float32.

### PR DESCRIPTION
**Pull request by ds-hwang**
https://github.com/apple/axlearn/pull/823 reveals this hidden bug.

In addition, add `lookup` API as downstream started abusing `_lookup` function. Actually this API is needed in inference.

### CODE REVIEW

**Comment by ds-hwang** [comment]
Could you review it? from #888

File: axlearn/common/quantizer_test.py [hunk 1]
```diff
@@ -414,6 +414,41 @@ def _loss(params, inputs, paddings, layer=layer):
...
+        inputs = np.random.rand(batch_size, seq_len, input_dim).astype(np.float32)
+        paddings = np.zeros((batch_size, seq_len)).astype(np.float32)
+        outputs, _ = F(
+            layer,
+            inputs=dict(inputs=inputs, paddings=paddings),
+            is_training=True,
+            prng_key=jax.random.PRNGKey(10),
+            state=layer_params,
+        )
+        self.assertEqual((batch_size, seq_len, num_groups), outputs.ids.shape)
```
[the remark this answers is not in the document]
Comment (ds-hwang) [reply axlearn/common/quantizer_test.py:439]: Good idea. Done.

File: axlearn/common/quantizer_test.py [hunk 2]
```diff
@@ -414,6 +414,42 @@ def _loss(params, inputs, paddings, layer=layer):
...
+        cfg = RandomVectorQuantizer.default_config().set(
+            name="test",
+            input_dim=input_dim,
+            codebook_dim=dim_from_all_codebooks // num_groups,
+            codebook_size=vocab_size,
+            num_codebooks=num_groups,
+        )
+        layer: RandomVectorQuantizer = cfg.instantiate(parent=None)
+        layer_params = layer.initialize_parameters_recursively(prng_key=jax.random.PRNGKey(1))
+        np.random.seed(2022)
```
Comment (markblee) [inline axlearn/common/quantizer_test.py:429]: > This is a convenience, legacy function that exists to support older code that uses the singleton RandomState.

Can we use `jax.random` instead?

**Review by markblee — COMMENTED** [review]
Thanks! Will approve once the internal PR is ready/passing.

**Review by ruomingp — APPROVED** [review]

**Pull request merged**